metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py

```diff
@@ -495,9 +552,55 @@ def get_entry_timestamp(entry: AuditLogEntry) -> datetime:
             yield entry
         logger.info(f"Finished loading {i} log entries from BigQuery")
 
+    def _create_operation_aspect_work_unit(
+        self, event: QueryEvent
+    ) -> Optional[MetadataWorkUnit]:
+        if event.statementType in OPERATION_STATEMENT_TYPES and event.destinationTable:
+            destination_table: BigQueryTableRef
+            try:
+                destination_table = event.destinationTable.remove_extras()
+            except Exception as e:
+                self.report.report_warning(
+                    str(event.destinationTable),
+                    f"Failed to clean up destination table, {e}",
+                )
+                return None
+            last_updated_timestamp: int = int(event.timestamp.timestamp() * 1000)
+            affected_datasets = []
+            if event.referencedTables:
+                for table in event.referencedTables:
+                    try:
+                        affected_datasets.append(
+                            _table_ref_to_urn(table.remove_extras(), self.config.env)
+                        )
+                    except Exception as e:
+                        self.report.report_warning(
+                            str(table),
+                            f"Failed to clean up table, {e}",
+                        )
+            operation_aspect = OperationClass(
```

Comment:
nit (can be addressed later): lines 581 - 598 can be refactored into a function in usage_common and all of the usage sources can just reuse it.